utilities/ps.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+from __future__ import annotations
+
+import subprocess
+from dataclasses import dataclass
+
+
+@dataclass
+class Service:
+    name: str
+    bindings: set[tuple[int, int]]
+
+    def print(self):
+        """
+        Print the formatted output for the service. The output contains the following:
+        - name (in bold)
+        - each URL (with the correct protocol) and the container port to which it maps
+        It specially handles the singular NGINX port 9443 which serves over ``https``.
+        """
+
+        print(f"\033[1m{self.name}:\033[0m")
+        for host_port, container_port in self.bindings:
+            proto = "http"
+            if self.name == "proxy" and container_port == 9443:
+                proto = "https"
+            print(f"- {proto:>5}://0.0.0.0:{host_port} (→ {container_port})")
+
+
+def get_ps() -> list[str]:
+    """
+    Invoke Docker Compose to get the configuration for all services. The config is
+    returned as a yaml string.
+
+    :return: the output printed by the subprocess to STDOUT
+    """
+
+    proc = subprocess.run(
+        [
+            "docker",
+            "ps",
+            "--filter",
+            "name=openverse",
```

Comment:
To disambiguate this, we can use `network=openverse_default` to select only containers running inside the Openverse docker-compose stack.